### PR TITLE
test: Tests started to fail after Datadog-dependency update

### LINKED ISSUE #225: Fix flaky/failing tests
Recently, builds have started to fail. Some tests are either failing constantly (merge was allowed even though a build was failing?) or sporadically (flaky test due to wrong test-set up).

### PR DESCRIPTION
**Pull request by bendiknesbo**
Closes #225

### CODE REVIEW

File: middleware/http/client_test.go [hunk 1]
```diff
@@ -102,5 +103,17 @@ func TestWrapClientW3C(t *testing.T) {
...
+	require.Equal(t, "dd", parts[0], "w3c tracestate list-member did not contain Datadog list-member")
+
+	// Assert Datadog-part of TraceState
+	parts = strings.Split(parts[1], ";")
+	sort.Strings(parts) // The Datadog-part of the Tracestate can be in any order.
+	require.Equal(t, 3, len(parts))
+	assert.Equal(t, "s:1", parts[0])
+	assert.Equal(t, "t.dm:-1", parts[1])
+	assert.True(t, strings.HasPrefix(parts[2], "t.tid:"))
+	assert.Equal(t, len("t.tid:65796a3f00000000"), len(parts[2]), "t.tid had invalid length.\nExample: %s\nGot:     %s", "t.tid:65796a3f00000000", parts[2])
```
Comment (arunpoudel) [inline middleware/http/client_test.go:118]: Should we say that the part after `:` is 128-bit instead?

Comment (arunpoudel) [reply middleware/http/client_test.go:118]: I am not sure how to achieve it though.

Comment (bendiknesbo) [reply middleware/http/client_test.go:118]: I was contemplating doing less validation at all, as the list-members are opaque strings unique and not publicly documented by Datadog

Comment (arunpoudel) [reply middleware/http/client_test.go:118]: Looks like this actually comes from the w3c spec: https://www.w3.org/TR/trace-context/#trace-id

Implement in dd-trace-go, here: https://github.com/DataDog/dd-trace-go/pull/2335

Comment (arunpoudel) [reply middleware/http/client_test.go:118]: Talked offline about it. The conclusion was we will ignore vendor specific tracestate header, [see](https://www.w3.org/TR/trace-context/#tracestate-header) for tests.

**Review by misund — APPROVED** [review]

**Review by arunpoudel — APPROVED** [review]

**Pull request merged**